addon/services/ajax.js

```diff
@@ -113,9 +113,9 @@ export default Ember.Service.extend({
         );
 
         if (response instanceof AjaxError) {
-          reject(response);
+          Ember.run.join(null, reject, response);
```

Comment:
@stefanpenner When writing tests that perform async operations, it's useful to be able to easily wait for all promises to resolve before finishing the test. @rwjblue create [`wait()`](https://github.com/switchfly/ember-test-helpers/blob/master/lib/ember-test-helpers/wait.js) for this purpose. This helper relies on the run loop to know if all scheduled timers have been cleared before resolving the `wait()`. Without this, we'd have to track of these promises inside of the service.